libshortfin/tests/api/array_test.py

```diff
@@ -121,3 +121,34 @@ def test_xtensor_types(scope, dtype, code, py_value, expected_str):
     r = repr(ary)
     print("__repr__ =", r)
     assert expected_str in r, f"Expected '{expected_str}' in '{r}'"
+
+
+@pytest.mark.parametrize(
+    "keys,expected_str",
+    [
+        # Simple indexing
+        ([0, 0], "{{0}}"),
+        # Row indexing
+        ([0], "{{0, 1, 2, 3}}"),
+        # Sliced indexing
+        ([1, slice(2, 4)], "{{2, 3}}"),
+        ([slice(1, 2), slice(2, 4)], "{{2, 3}}"),
+    ],
+)
+def test_view(scope, keys, expected_str):
+    src = sfnp.device_array.for_host(scope.device(0), [4, 4], sfnp.uint8)
+    src.fill(b"\0\1\2\3")
```

Comment:
Actually, I misspoke: `fill` is asynchronous, so you are observing the value before the queue has processed the commands. This needs some more ergonomic work: we generally don't want to be async filling a host buffer like this for this kind of case. And I'm missing synchronous waits in the Python API, which would be the most convenient here.

Easiest thing would be if the items= API was available.

Best thing to do for now is to write this async like the tests above. Look at `test_fill_copy_to_for_transfer`. Note the `await device` before observing the result.

There will be better ergonomics for this eventually but no need to wait for them for a test.

Reply:
ah, added an await to fix it.